### PR TITLE
use DefaultAzureCredential if login not provided for Data Factory

### LINKED ISSUE #19077: Add support for more authentication types in Azure Data Factory hook
### Description

At the moment the Azure Data Factory hook only supports `TokenCredential` (service principal client ID and secret). It would be very useful to have support for other authentication types like managed identity. Preferably, we would create a `DefaultTokenCredential` if the client ID and secret are not provided in the connection.

### Use case/motivation

We're using Airflow on Azure Kubernetes Service and this would allow us to use the pod identity for authentication which is a lot cleaner than creating a service principal.

### Related issues

_No response_

### Are you willing to submit a PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

### PR DESCRIPTION
**Pull request by sdebruyn**
closes: #19077 

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/main/UPDATING.md).

### CODE REVIEW

File: airflow/providers/microsoft/azure/hooks/data_factory.py [hunk 1]
```diff
@@ -144,10 +144,14 @@ def get_conn(self) -> DataFactoryManagementClient:
...
 
-        self._conn = DataFactoryManagementClient(
-            credential=ClientSecretCredential(
+        credential = DefaultAzureCredential()
+        if conn.login is not None and conn.password is not None:
+            credential = ClientSecretCredential(
                 client_id=conn.login, client_secret=conn.password, tenant_id=tenant
-            ),
+            )
+
```
Comment (eladkal) [inline airflow/providers/microsoft/azure/hooks/data_factory.py]:
```suggestion

```

**Review by eladkal — CHANGES_REQUESTED** [review]

File: airflow/providers/microsoft/azure/hooks/data_factory.py (hunk 1 above)
Comment (eladkal) [reply airflow/providers/microsoft/azure/hooks/data_factory.py]: to fix static tests

File: docs/apache-airflow-providers-microsoft-azure/connections/adf.rst [hunk 2]
```diff
@@ -27,11 +27,14 @@ The Microsoft Azure Data Factory connection type enables the Azure Data Factory
 Authenticating to Azure Data Factory
 ------------------------------------
 
-There is one way to connect to Azure Data Factory using Airflow.
+There are multiple ways to connect to Azure Data Factory using Airflow.
 
 1. Use `token credentials
    <https://docs.microsoft.com/en-us/azure/developer/python/azure-sdk-authenticate?tabs=cmd#authenticate-with-token-credentials>`_
    i.e. add specific credentials (client_id, secret, tenant) and subscription id to the Airflow connection.
+1. Fallback on `DefaultAzureCredential
```
Comment (josh-fell) [inline docs/apache-airflow-providers-microsoft-azure/connections/adf.rst]: This item should be number 2 in the list.

**Comment by sdebruyn** [comment]
Thank you for assisting me with this PR! It's my first Airflow PR and I have a hard time setting the repo up locally. If you wouldn't mind, feel free to reach out to me on Slack to help me to set it up. It will save me time with future contributions.

File: airflow/providers/microsoft/azure/hooks/data_factory.py [hunk 3]
```diff
@@ -144,10 +144,13 @@ def get_conn(self) -> DataFactoryManagementClient:
...
-        self._conn = DataFactoryManagementClient(
-            credential=ClientSecretCredential(
+        credential = DefaultAzureCredential()
+        if conn.login is not None and conn.password is not None:
+            credential = ClientSecretCredential(
                 client_id=conn.login, client_secret=conn.password, tenant_id=tenant
-            ),
+            )
+        self._conn = DataFactoryManagementClient(
+            credential=credential,
```
Comment (eladkal) [inline airflow/providers/microsoft/azure/hooks/data_factory.py]: Can we test this?
https://github.com/apache/airflow/blob/main/tests/providers/microsoft/azure/hooks/test_azure_data_factory.py

**Review by eladkal — CHANGES_REQUESTED** [review]

File: airflow/providers/microsoft/azure/hooks/data_factory.py (hunk 3 above)
Comment (sdebruyn) [reply airflow/providers/microsoft/azure/hooks/data_factory.py]: How do I run these?

Comment (josh-fell) [reply airflow/providers/microsoft/azure/hooks/data_factory.py]: Check out [this doc](https://github.com/apache/airflow/blob/main/TESTING.rst#airflow-unit-tests). It details how to run unit tests locally within a couple IDEs or Breeze/virtualenv.

Comment (sdebruyn) [reply airflow/providers/microsoft/azure/hooks/data_factory.py]: @eladkal @josh-fell test added :) thank you for your patience with my first PR!

**Review by potiuk — APPROVED** [review]

**Comment by potiuk** [comment]
@eladkal WDYT?

**Review by eladkal — APPROVED** [review]

**Pull request merged**